specification/dartLangSpec.tex

```diff
@@ -1317,127 +1350,256 @@ \section{Variables}
-\code{\DYNAMIC{} \GET{} $v$}
+
+\begin{itemize}
+\item \code{\STATIC?\,\,\LATE?\,\,\VAR\,\,\id;}
+\item \code{\STATIC?\,\,\LATE?\,\,\VAR\,\,\id{} = $e$;}
+\item \code{\STATIC?\,\,\LATE?\,\,\FINAL\,\,\id;}
+\item \code{\STATIC?\,\,\LATE?\,\,\FINAL\,\,\id{} = $e$;}
+\item \code{\STATIC?\,\,\CONST\,\,\id{} = $e$;}
+\end{itemize}
+
+\noindent
+implicitly induces a getter with the header
+\code{\DYNAMIC\,\,\GET\,\,\id},
 whose invocation evaluates as described below
 (\ref{evaluationOfImplicitVariableGetters}).
-%% TODO[inference]: We assume inference has taken place, i.e., inferred types
-%% are written explicitly. Does this mean that the initialized variants
-%% cannot exist (not even for `$e$` of type `dynamic`?). We probably don't
-%% want to start talking about a grammar before inference and another one
-%% after inference.
-In these cases, the static type of $v$ is \DYNAMIC{}
+In these cases, the declared type of \id{} is \DYNAMIC{}
 (\ref{typeDynamic}).
 
+\commentary{%
+Type inference is assumed to have taken place already
+(\ref{overview}).
+For example, an instance variable declaration of the form
+\code{\VAR\,\,x;} could have been transformed into
+\code{$T$\,\,x;} based on member signatures named \code{x} in superinterfaces
+of the enclosing class, and
+\code{\VAR\,x\,\,=\,\,$e$;} could have been transformed into
+\code{$T$\,x\,\,=\,\,$e$;}
+where $T$ is the static type of $e$.
+In that case, the type is $T$ rather than \DYNAMIC.%
+}
+\EndCase
+
+\LMHash{}%
+\Case{Setter: Mutable variable with declared type}
+A mutable variable declaration of one of the forms
+
+\begin{itemize}
+\item \code{\STATIC?\,\,\LATE?\,\,$T$\,\,\id;}
+\item \code{\STATIC?\,\,\LATE?\,\,$T$\,\,\id{} = $e$;}
+\end{itemize}
+
+\noindent
+implicitly induces a setter (\ref{setters}) with the header
+\code{\VOID\,\,\SET\,\,\id($T$\,\,$x$)},
+whose execution sets the value of \id{} to the incoming argument $x$.
+
 \LMHash{}%
+\Case{Setter: Mutable variable with no declared type, with initialization}
 A mutable variable declaration of the form
-\code{{} $T$ $v$;}
-or \code{$T$ $v$ = $e$;}
-induces an implicit setter function (\ref{setters}) with signature
-\code{\VOID{} \SET{} $v$=($T$ $x$)}
-whose execution sets the value of $v$ to the incoming argument $x$.
+\code{\STATIC?\,\,\LATE?\,\,\VAR\,\,\id{} = $e$;}
+implicitly induces a setter with the header
+\code{\VOID\,\,\SET\,\,\id(\DYNAMIC\,\,$x$)},
+whose execution sets the value of \id{} to the incoming argument $x$.
+
+\commentary{%
+Type inference could have provided a type different from \DYNAMIC{}
+(\ref{overview}).%
+}
+\EndCase
 
 \LMHash{}%
+\Case{Setter: Mutable variable with no declared type, no initialization}
 A mutable variable declaration of the form
-\code{\VAR{} $v$;}
-or \code{\VAR{} $v$ = $e$;}
-induces an implicit setter function with signature
-\code{\VOID{} \SET{} $v$=(\DYNAMIC{} $x$)}
-whose execution sets the value of $v$ to the incoming argument $x$.
+\code{\STATIC?\,\,\LATE?\,\,\VAR\,\,\id;}
```

Comment:
Should we include a non-local `static? late final id;`?

(Since it really is mutable.)

Otherwise I can see it's in line 1556.